packages/core/src/shared/errors.ts

```diff
@@ -886,28 +875,59 @@ function isError(err: Error, id: string, messageIncludes: string = '') {
 }
 
 /**
- * AWS SDK clients may rarely make requests that results in something other than JSON data
- * being returned (e.g html). This will cause the client to throw a SyntaxError as a result
+ * AWS SDK clients make requests with the expected result to be JSON data.
+ * But in some cases the request may fail and result in an error HTML page being returned instead
+ * of the JSON. This will cause the client to throw a `SyntaxError` as a result
  * of attempt to deserialize the non-JSON data.
- * While the contents of the response may contain sensitive information, there may be a reason
- * for failure embedded. This function attempts to extract that reason.
  *
- * Example error message before extracting:
- * "Unexpected token '<', "<html><bod"... is not valid JSON Deserialization error: to see the raw response, inspect the hidden field {error}.$response on this object."
+ * But within the `SyntaxError` instance is the real reason for the failure.
+ * This class attempts to extract the underlying issue from the SyntaxError.
  *
- * If the reason cannot be found or the error is not a SyntaxError, return undefined.
+ * Example SyntaxError message before extracting the underlying issue:
+ * "Unexpected token '<', "<html><bod"... is not valid JSON Deserialization error: to see the raw response, inspect the hidden field {error}.$response on this object."
  */
-export function getReasonFromSyntaxError(err: Error): string | undefined {
-    if (!(err instanceof SyntaxError)) {
-        return undefined
+export class SDKClientResponseError extends ToolkitError {
```

Comment:
"sdk" could mean anything

```suggestion
export class AwsClientResponseError extends ToolkitError {
```